From 48dab3b60a2b0b690c8bbfb155335e8e5423b647 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Wed, 11 Dec 2024 14:48:50 +0100
Subject: [PATCH 761/954] feat(st-clk): add ck_bus_risaf4 clock for STM32MP2

Add management of ck_bus_risaf4 clock.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git b864d98a5273fffd273b043e4722d3709ae6bc53 feat(st-clk): add ck_bus_risaf4 clock for STM32MP2]
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Change-Id: I6ac2aff07484bfc22210ee9d3e46a97d1735f34b
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/424220
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 drivers/st/clk/clk-stm32mp2.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/st/clk/clk-stm32mp2.c b/drivers/st/clk/clk-stm32mp2.c
index 65a9c877dc..ec6c05104e 100644
--- a/drivers/st/clk/clk-stm32mp2.c
+++ b/drivers/st/clk/clk-stm32mp2.c
@@ -295,6 +295,7 @@ enum clock {
 	_CK_USB3PCIEPHY,
 	_CK_USBTC,
 #endif /* !STM32MP21 */
+	_CK_BUS_RISAF4,
 
 	CK_LAST
 };
@@ -1618,7 +1619,8 @@ static const struct clk_stm32 stm32mp2_clk[CK_LAST] = {
 	STM32_GATE(_CK_GPIOZ, CK_BUS_GPIOZ, _CK_ICN_LS_MCU, 0, GATE_GPIOZ),
 	STM32_GATE(_CK_RTC, CK_BUS_RTC, _CK_ICN_LS_MCU, 0, GATE_RTC),
 
-	STM32_GATE(_CK_DDRCP, CK_BUS_DDR, _CK_ICN_DDR, 0, GATE_DDRCP),
+	STM32_GATE(_CK_BUS_RISAF4, CK_BUS_RISAF4, _CK_ICN_LS_MCU, CLK_IS_CRITICAL, GATE_DDRCP),
+	STM32_GATE(_CK_DDRCP, CK_BUS_DDR, _CK_ICN_DDR, CLK_IS_CRITICAL, GATE_DDRCP),
 
 	/* WARNING 2 CLOCKS FOR ONE GATE */
 #if STM32MP21
-- 
2.53.0

